paasta_tools/paasta_cli/cmds/rollback.py

```diff
@@ -53,22 +54,47 @@ def add_subparser(subparsers):
     list_parser.set_defaults(command=paasta_rollback)
 
 
+def validate_given_instances(service, args_instances):
+    """Trying to find which instances are actually valid for the given service"""
+    service_instances = list_all_instances_for_service(service)
```

Comment:
I think you should move this logic to the caller; it'd be better if this was just a wrapper function around the set intersection/difference.